prov/efa/src/rdm/efa_rdm_pke.h

```diff
@@ -207,30 +179,56 @@ struct efa_rdm_pke {
 	uint32_t flags;
 
 	/**
-	 * @brief link multiple MEDIUM RTM with same message ID together
+	 * @brief link multiple MEDIUM/RUNTREAD RTM with same
+	 * message ID together
 	 *
 	 * @details
 	 * used on receiver side only
 	 */
 	struct efa_rdm_pke *next;
 
-	/** @brief information of send buffer
-	 *  @todo make this field a union with `next` to reduce memory usage
+	/**
+	 * @brief payload buffer
+	 * 
+	 * @details
+	 * "payload" refers to a piece of user data that is
+	 * going over wire.
+	 *
+	 * the pointer "payload" can be directly pointing to
+	 * user's buffer or provider's internal bounce
+	 * buffer (where user data has been copied to).
 	 */
```

Comment:
TBH this makes payload even more confusing.... do you mean it is actually `localdata`?

payload typically means the same thing as `wiredata`, which is the actual data sent over the network.

Reply:
> payload typically means the same thing as wiredata, which is the actual data sent over the network.

Yes. "payload" is the actual data that is going over wire.

EFA device support 2 iov, which means we can send 2 buffers that are not continuous in 1 packet.

wiredata consume 1st iov, which always have the header.

payload consumes 2nd iov, which has the user's data. Sometimes it is pointing directly to user's buffer.

What kind of wording would you suggest to make it more clear?

Reply:
I updated the comment after our offline discussion. Please take a look